test/invariant/fuzz/properties/PropertiesAllo.t.sol

```diff
@@ -1,66 +1,345 @@
+
+        bytes32 _poolManagerRole = bytes32(_poolId);
+
+        (bool _success,) =
+            targetCallDefault(address(allo), 0, abi.encodeWithSelector(allo.addPoolManagers.selector, _poolId, _managers));
+
+        if (msg.sender == _admin) {
+            if (_success) {
+                assertTrue(allo.hasRole(_poolManagerRole, _newManager), "addPoolManagers failed");
+                ghost_poolManagers[_poolId].push(_newManager);
+            } else {
+                assertTrue(_newManager == address(0), "addPoolManager failed");
+            }
+        } else {
+            if (_success) {
+                fail("addPoolManager only admin should be able to add managers");
+            }
+        }
+    }
+
+    ///@custom:property-id 11-b
+    ///@custom:property pool admin can always remove pool managers
+    function prop_poolAdminCanAlwaysRemoveManagers(uint256 _idSeed, uint256 _managerIndex) public {
+        _idSeed = bound(_idSeed, 0, ghost_poolIds.length - 1);
+        uint256 _poolId = ghost_poolIds[_idSeed];
+        _managerIndex = bound(_managerIndex, 0, ghost_poolManagers[_poolId].length - 1);
+        address _admin = ghost_poolAdmins[_poolId];
+        address[] storage _managers = ghost_poolManagers[_poolId];
+        address _manager = _managers[_managerIndex];
+        bytes32 _poolManagerRole = bytes32(_poolId);
+
+        address[] memory _removeManagers = new address[](1);
+        _removeManagers[0] = _manager;
+
+        (bool _success,) = targetCallDefault(
+            address(allo), 0, abi.encodeWithSelector(allo.removePoolManagers.selector, _poolId, _removeManagers)
+        );
+
+        if (msg.sender == _admin) {
+            if (_success) {
+                assertTrue(!allo.hasRole(_poolManagerRole, _manager), "removePoolManagers failed");
+                delete _managers[_managerIndex];
+            } else {
+                assertTrue(_manager == address(0), "removePoolManager failed");
+            }
+        } else {
+            if (_success) {
+                fail("removePoolManager only admin should be able to remove managers");
+            }
+        }
+    }
+
+    ///@custom:property-id 12
+    ///@custom:property pool manager can always withdraw within strategy limits/logic
+
+    ///@custom:property-id 13
+    ///@custom:property pool manager can always change metadata
+    function prop_poolManagerCanAlwaysChangeMetadata(uint256 _idSeed, uint256 _seedManager, Metadata calldata _metadata)
+        public
+    {
+        _idSeed = bound(_idSeed, 0, ghost_poolIds.length - 1);
+        uint256 _poolId = ghost_poolIds[_idSeed];
+        uint256 _managerIndex = _seedManager % (ghost_poolManagers[_poolId].length - 1);
+        address _manager = ghost_poolManagers[_poolId][_managerIndex];
+
+        (bool _success,) =
+            targetCallDefault(address(allo), 0, abi.encodeWithSelector(allo.updatePoolMetadata.selector, _poolId, _metadata));
+
+        if (msg.sender == _manager) {
```

Comment:
we pick *a random* manager, then use msg.sender, then compare this manager to the sender -> if another (valid) manager is the msg.sender, then the test fails (while it shouldn't), like it does in CI

Reply:
good catch this was outdated with my previous implementation, now I will just use the `msg.sender` and then check if it was or not a manager, depending on that will run the assertions